pymongo/message.py

```diff
@@ -798,7 +798,7 @@ def write_command(self, cmd, request_id, msg, docs):
             self._start(cmd, request_id, docs)
             start = datetime.datetime.now()
         try:
-            reply = self.sock_info.write_command(request_id, msg)
+            reply = self.sock_info.write_command(request_id, msg, self.codec)
```

Comment:
We need a similar change in _EncryptedBulkWriteContext.execute().

Reply:
I wasn't sure how to test that locally, waiting for evergreen